lib/simulator/simulatormodule.c

```diff
@@ -1024,7 +1039,10 @@ static PyObject *deregister_callback(PyObject *self, PyObject *args)
     gstate = TAKE_GIL();
 
     pSihHdl = PyTuple_GetItem(args, 0);
-    hdl = (gpi_sim_hdl)PyLong_AsLong(pSihHdl);
+    if (!gpi_sim_hdl_converter(pSihHdl, &hdl)) {
```

Comment:
May as well rewrite this to `if (!PyArg_ParseTuple(args, "O&", gpi_sim_hdl_converter, &hdl)`, which will check the number of arguments too